pkg/analysis/conflictingmarkers/analyzer.go

```diff
@@ -68,25 +68,36 @@ func (a *analyzer) run(pass *analysis.Pass) (any, error) {
 	}
 
 	inspect.InspectFields(func(field *ast.Field, stack []ast.Node, _ extractjsontags.FieldTagInfo, markersAccess markers.Markers) {
-		checkField(pass, field, markersAccess, a.conflictSets)
+		var structName string
+		// Find the struct name from the stack
+		for i := len(stack) - 2; i >= 0; i-- {
+			if typeSpec, ok := stack[i].(*ast.TypeSpec); ok {
+				if _, ok := typeSpec.Type.(*ast.StructType); ok {
+					structName = typeSpec.Name.Name
+					break
+				}
+			}
+		}
+
+		checkField(pass, field, structName, markersAccess, a.conflictSets)
 	})
 
 	return nil, nil //nolint:nilnil
 }
 
-func checkField(pass *analysis.Pass, field *ast.Field, markersAccess markers.Markers, conflictSets []ConflictSet) {
+func checkField(pass *analysis.Pass, field *ast.Field, structName string, markersAccess markers.Markers, conflictSets []ConflictSet) {
 	if field == nil || len(field.Names) == 0 {
 		return
 	}
 
 	markers := utils.TypeAwareMarkerCollectionForField(pass, markersAccess, field)
 
 	for _, conflictSet := range conflictSets {
-		checkConflict(pass, field, markers, conflictSet)
+		checkConflict(pass, field, structName, markers, conflictSet)
 	}
 }
 
-func checkConflict(pass *analysis.Pass, field *ast.Field, markers markers.MarkerSet, conflictSet ConflictSet) {
+func checkConflict(pass *analysis.Pass, field *ast.Field, structName string, markers markers.MarkerSet, conflictSet ConflictSet) {
```

Comment:
Will this handle conflicting tags when applied to a tyoe definition (e.g. a string alias)?